### PR TITLE
fix: avoid panics in match case diagnostic

### LINKED ISSUE #8809: thread '<unnamed>' panicked at 'internal error: entered unreachable code: Replacement
image

RA crashed while working on https://github.com/parallel-finance/parallel, I just did:

```
vim runtime/parallel/src/lib.rs
```

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

closes #8809

### CODE REVIEW

File: crates/hir_ty/src/diagnostics/decl_check.rs [hunk 1]
```diff
@@ -150,29 +150,13 @@ impl<'a, 'b> DeclValidator<'a, 'b> {
...
-
         // Check the patterns inside the function body.
+        // This includes function parameters.
         let pats_replacements = body
             .pats
             .iter()
-            // We aren't interested in function parameters, we've processed them above.
-            .filter(|(pat_idx, _)| !body.params.contains(&pat_idx))
+            // // We aren't interested in function parameters, we've processed them above.
+            // .filter(|(pat_idx, _)| !body.params.contains(&pat_idx))
```
Comment (lnicola) [inline crates/hir_ty/src/diagnostics/decl_check.rs:159]: Do we want to keep this?

**Comment by lnicola** [comment]
```
thread 'tests::test_attr_macro' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 32, kind: Other, message: "The process cannot access the file because it is being used by another process." }', crates\proc_macro_srv\src\dylib.rs:209:30
thread 'tests::test_derive_empty' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: Error("Could not read file magic") }', crates\proc_macro_srv\src\tests\utils.rs:56:48
```

File: crates/hir_ty/src/diagnostics/decl_check.rs (hunk 1 above)
Comment (matklad) [reply crates/hir_ty/src/diagnostics/decl_check.rs]: good catch

**Comment by matklad** [comment]
bors retry

**Comment by matklad** [comment]
bors r+

huh

**Comment by jonas-schievink** [comment]
bors retry

hopefully https://github.com/rust-analyzer/rust-analyzer/pull/9086 fixed this

**Pull request merged**